src/classes/database.ts

```diff
@@ -78,141 +101,126 @@ export class Database {
   private async initSettings() {
     const collection = this.collection<ServerInfo>(Collections.SERVER_INFO);
     let settings = await collection.findOneAsync({});
-    if (settings == null) {
+    
+    if (!settings) {
       Logger.log("Generating new JWT secret");
       settings = {
         JWT_SECRET: crypto.randomBytes(64).toString("hex"),
         version: CURRENT_VERSION,
         currentEvent: SeasonalEvents.SET_NoSeasonalEvent,
       };
-      collection.insertAsync(settings).catch(Logger.log);
+      await handleDatabaseError(() => collection.insertAsync(settings));
     }
     this.token = settings.JWT_SECRET;
 
     if (!settings.currentEvent) {
-      await collection.updateAsync(
-        {},
-        { $set: { currentEvent: SeasonalEvents.SET_NoSeasonalEvent } }
-      );
+      await updateCollection(collection, {}, { currentEvent: SeasonalEvents.SET_NoSeasonalEvent });
     }
+
     await this.checkVersionAndMigrations(settings.version);
   }
 
   private async checkVersionAndMigrations(version: number) {
-    // Switch without breaks because migrations should be secuencial and cummulative
-    switch (version) {
-      default: // If version was pre-0
-        await this.DLCCharactersFix();
-      case 0: // if version was pre-1 (you get the idea)
-        await this.removeTrophiesFix();
-      case 1:
-        this.reduceCommonDatabase();    
-        break;
-      case CURRENT_VERSION:
-        break;
+    Logger.log(`Running migrations for version ${version}`);
+    
+    const migrationTasks = [
+      { version: 0, task: () => this.DLCCharactersFix() },
+      { version: 1, task: () => this.removeTrophiesFix() }
+    ];
+
+    for (const { version: v, task } of migrationTasks) {
+      if (version <= v) await task();
     }
-    await this.collection<ServerInfo>(Collections.SERVER_INFO).updateAsync(
+
+    await updateCollection(
+      this.collection<ServerInfo>(Collections.SERVER_INFO),
       {},
-      { $set: { version: CURRENT_VERSION } }
+      { version: CURRENT_VERSION }
```

Comment:
If you specify an object instead of a modifier (`$set`) you're asking NEDB to change the whole document instead of a property, so this would probably lose other properties of the collection (like the JWT token)